src/reusableState/customList/tests/StringListReducers.spec.ts

```diff
@@ -0,0 +1,142 @@
+import {IReduxAction} from '../../../utils/ReduxUtils';
+import {
+    addStringList,
+    addValueStringList,
+    IStringListPayload,
+    removeStringList,
+    removeValueStringList,
+    StringListActions,
+} from '../StringListActions';
+import {IStringListCompositeState, stringListCompositeState, stringListInitialState} from '../StringListReducers';
+
+describe('StringList', () => {
+
+    describe('StringList Reducers', () => {
+
+        const stateId: string = 'test';
+        const oldState: IStringListCompositeState = {
+            [stateId]: {
+                id: stateId,
+                list: [],
+            },
+        };
+
+        const genericAction: IReduxAction<IStringListPayload> = {
+            type: 'pokeball',
+        };
+
+        it('should return the default state if the action has no payload and the state is undefined', () => {
+            const newState: IStringListCompositeState = stringListCompositeState(undefined, genericAction);
+
+            expect(newState).toEqual(stringListInitialState);
+            expect(newState).not.toBe(stringListInitialState);

```

Comment:
`expect(newState).not.toBe(stringListInitialState);` doesn't have to be, it could actually be the same object if the state was not modified after an action.